mantidproject/mantid
# ISIS Powder GEM added save all flag

**Description of work.**
Removed save_angles flag as it was always set via texture mode. Added a save_all flag that allows disabling of saving out .NXS files since these are often not needed in texture mode, and can take up a lot of space. by default save_all will be set to true to avoid affecting scripts that already exist.

**To test:**
To test run the following code
```
from isis_powder import Gem

config_file_path = r"Path\to\Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path,user_name="test",texture_mode=True, save_all=False)
Gem_example.create_vanadium(first_cycle_run_no=84313)
Gem_example.focus(run_number=84313,input_mode="Individual")
```
with these [files](https://github.com/mantidproject/mantid/files/2296902/Test.zip) (you will need to update the paths both in the script and in the config file.)
then check the output folder, it should not contain a .NXS file or .gsas file, then run
```
from isis_powder import Gem

config_file_path = r"Path\to\Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path,user_name="test",texture_mode=True)
Gem_example.create_vanadium(first_cycle_run_no=84313)
Gem_example.focus(run_number=84313,input_mode="Individual")
```
or set `save_all=True` either of these inputs should generate .NXS and .gsas files. 
Fixes #23279


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
